### PR TITLE
fix dynamic space direction (#1999)

### LINKED ISSUE #1999: D/F Switching is no longer working
According to the direction in which the user is tracing, the direction of the `f` and `d` (or `Space` and `Shift + Space`) keys should dynamically switch, so that pressing `f` (or `Space`) always moves in the tracing direction.
This functionality no longer works.

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/1999/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by daniel-wer**
My PR that fixed the rotation of the view in flight mode, broke the dynamic space direction, because the `setRotationAction` was "misused" to affect the dynamic space direction. I made this more explicit by introducing a new `setDirectionAction` that is used instead.

### Steps to test:
- Press `Space` in orthogonal mode and note the direction in which the z-Value of the position changes. Now set a node, then move backwards using `Space + Shift` and set another node. Now press `Space` again - the direction in which the z-Value of the position changes should be reversed.
- Yes this is a somewhat obscure feature, but it has been there since the beginning and it seems to be quite essential to "advanced" tracers. :)

### Issues:
- fixes #1999 

------
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/oxalis/model/actions/actions.js [hunk 1]
```diff
@@ -18,7 +19,8 @@ export type ActionType =
   | TaskActionType
   | SaveActionType
   | ViewModeActionType
-  | AnnotationActionTypes;
+  | AnnotationActionTypes
+  | FlycamActionType;
```
Comment (philippotto) [inline app/assets/javascripts/oxalis/model/actions/actions.js]: Good catch! It's a bummer that flow didn't error before, apparently? I guess the dispatch typing is not good enough for that?

File: app/assets/javascripts/test/reducers/flycam_reducer.spec.js [hunk 2]
```diff
@@ -155,7 +155,7 @@ test("Flycam should move in ortho mode", t => {
 });
 
 test("Flycam should move in ortho mode with dynamicSpaceDirection", t => {
-  let newState = FlycamReducer(initialState, FlycamActions.setRotationAction([0, 0, -2]));
```
Comment (philippotto) [inline app/assets/javascripts/test/reducers/flycam_reducer.spec.js]: The test caption suggests that `dynamicSpaceDirection` is tested, but the tests didn't fail? Is it easy to add a new test which covers the scenario that you are fixing?

File: app/assets/javascripts/oxalis/model/actions/actions.js (hunk 1 above)
Comment (daniel-wer) [reply app/assets/javascripts/oxalis/model/actions/actions.js]: I think the problem is that if flow doesn't know a specific action type `"MY_ACTION"` and it sees a `case "MY_ACTION":` it stops type checking as it thinks this case is never going to be executed, right?
Would be nice if flow complained when it encounters a non-reachable case. (if I understand this correctly)

**Comment by daniel-wer** [comment]
Good point, I'd like to add a test as well, ideally one which calls `setWaypoint` directly, but I can't think of a nice and easy way to do this (as the `SkeletonTracingPlaneController` is a React component and I don't have a reference to it). Would this be easier using the testing stuff you're investigating currently?

**Comment by daniel-wer** [comment]
@philippotto Any idea regarding the testing?

**Comment by philippotto** [comment]
> Good point, I'd like to add a test as well, ideally one which calls setWaypoint directly, but I can't think of a nice and easy way to do this (as the SkeletonTracingPlaneController is a React component and I don't have a reference to it). Would this be easier using the testing stuff you're investigating currently?

Right, that's not really straightforward. Enzyme could be used for mounting this controller component and hooking it up to a store, but that might be a bit of an overkill. I'm having problems to mount the tracing view with enzyme, anyway, so, I'd suggest to leave the tests as is for now :)

**Pull request merged**